src/functions_framework/__init__.py

```diff
@@ -68,18 +72,44 @@ def view_func(path):
     return view_func
 
 
-def _is_binary_cloud_event(request):
-    return (
-        request.headers.get("ce-type")
-        and request.headers.get("ce-specversion")
-        and request.headers.get("ce-source")
-        and request.headers.get("ce-id")
-    )
+def _run_cloudevent(function, request):
+    data = request.get_data()
+    event = from_http(data, request.headers)
+    function(event)
+
+
+def _cloudevent_view_func_wrapper(function, request):
+    def view_func(path):
+        try:
+            _run_cloudevent(function, request)
+        except (
+            cloud_exceptions.CloudEventMissingRequiredFields,
+            cloud_exceptions.CloudEventTypeErrorRequiredFields,
+        ):
+            flask.abort(
+                400,
+                description=(
+                    "Function was defined with FUNCTION_SIGNATURE_TYPE=cloudevent "
```

Comment:
Can we be more specific about what the problem is? It will save troubleshooting cycles to simply list the missing fields or the type errors. Failing this, can we deep link to the CloudEvent spec for the required fields?

Reply:
cloud_exceptions errors should have details on which field was missing. I included the cloud_exceptions error in the flask abort description.